doc/releases/migration-guide-4.1.rst

```diff
@@ -524,6 +524,15 @@ LoRa
   additional ``user_data`` parameter, which is a void pointer. This parameter can be used to reference
   any user-defined data structure. To maintain the current behavior, set this parameter to ``NULL``.
 
+State Machine Framework
+=======================
+
+* State run actions now return an :c:enum:`smf_state_result` value instead of void.
+  :c:func:`smf_set_handled` has been removed, and the return code determines if the
+  event is propagated to parent run actions or has been handled in this run action.
```

Comment:
I suggest flipping the order here - mention that `smf_set_handled` has been removed first and then describe the changes to state run actions.
It would also be helpful to provide instructions on how to transition an existing hierarchical state machine - something like "run states called smf_set_handled() should return SMF_EVENT_HANDLED and otherwise should return SMF_EVENT_PROPAGATE."